multiwallet_test.go

```diff
@@ -0,0 +1,31 @@
+package dcrlibwallet_test
+
+import (
+	"os"
+	"testing"
+	"testing/quick"
+
+	. "github.com/raedahgroup/dcrlibwallet"
+)
+
+// canUseDir returns true if the program can create the directory
+// It will create the directory if it can
+func canUseDir(directory string) bool {
+	return os.MkdirAll(directory, os.ModePerm) == nil
+}
+
+// TestNewMultiWalletPath checks that NewMultiWallet returns an error for invalid
+// paths or no error for valid paths
+func TestNewMultiWallet(t *testing.T) {
+	f := func(rootDir string) bool {
+		canUse := canUseDir(rootDir)
+		_, err := NewMultiWallet(rootDir, "", "testnet")
+		if canUse {
+			return err == nil
+		}
+		return err != nil
+	}
```

Comment:
As have been said previously, this test is buggy. I made some changes to the `f` function, ran the test a couple times to get 2 set of results (one passed, one failed) which reveal some issues with this testing code. The log outputs displayed for both results are truncated.

#### Modified the `f` function to print somewhat detailed reports
```
f := func(rootDir string) bool {
	canUse := canUseDir(rootDir)
	_, err := NewMultiWallet(rootDir, "", "testnet")

	if canUse {
		passed := err == nil

		var report string
		if passed {
			report = fmt.Sprintf("GOOD: %U is a valid path and we got no errors. Yay!", rootDir)
		} else {
			report = fmt.Sprintf("BAD: %U is a valid path but we got an error: %s.", rootDir, err.Error())
		}
		println(report)

		return passed
	}

	passed := err != nil

	var report string
	if passed {
		report = fmt.Sprintf("GOOD: %U is NOT a valid path, so we got an error: %s.", rootDir, err.Error())
	} else {
		report = fmt.Sprintf("BAD: %U is NOT a valid path but we got no errors. Weird!", rootDir)
	}
	println(report)

	return passed
}
```

#### First test result + log output (TEST PASSED with exit code 0, but should it have?)
```
GOOD: "\U000cbb31\U0006ecc1⋯\U000859ac𮗚\U0008311f\U000df8a3\U00076647" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U0008ddd5𡢙\U000c97e4\U00106c29\U000ac4b7̌\U000f4f23\U0004faa2\U000547ee\U0003e399\U00046378\U0001f269\U0010a688𬡚\U0009449d𗢷" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U0001ecac\U00095bba奕\U000782e0\U000ceb5b\U000b03f1\U0007d731\U00092766𨞈\U000a9ae0\U000aebf2" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U00066d7c\U000c02a8" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U0005214c\U00075f48\U0004562f\U0004ee05" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U00054699\U00097366\U0004709a" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U00085c61\U00042df4\U000149ee\U0006f80c\U000ada6e\U00067402" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U00050e5a\U00043f9f贉\U0007b60f\U0005f314\U000bd61d\U0005847b�\U000622ce\U000671f0췢\U0010a726\U000e4da9" is NOT a valid path, so we got an error: invalid net type.

Process finished with exit code 0
```
Analysis
- All test parameters passed to `f` were invalid paths. So it is expected that `NewMultiWallet` will return an error; and it did! But the error returned was `invalid net type` instead of a `PathError`. Agreed, if the netType was indeed valid, the error would have been a `PathError`, but this just goes to emphasize that the `NewMultiWallet` function can return an error whether or not a valid path is provided.
- If a valid netType was passed, the error would look like `GOOD: "\U0007af92\U0010ac8d\U0009d6ae" is NOT a valid path, so we got an error: failed to create rootDir: mkdir 񺾒􊲍򝚮: illegal byte sequence.` This kind of error should be what is used in determining whether the test should pass or fail. Any error shouldn't do.

#### Second test result + log output (TEST FAILED with exit code 1, but should it have?)
```
GOOD: "" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U000cd531\U00030f92\U0006d2cf\U000f2b31" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U001018f1\U000c57d4𨃁\U00101e3f\U00107546" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U000d9d16" is NOT a valid path, so we got an error: invalid net type.
GOOD: "" is NOT a valid path, so we got an error: invalid net type.
GOOD: "" is NOT a valid path, so we got an error: invalid net type.
GOOD: "𗰶𨺈䵄\U000eca4a\U000c2b42" is NOT a valid path, so we got an error: invalid net type.
GOOD: "\U0006d593𤣯\U000f1629" is NOT a valid path, so we got an error: invalid net type.
BAD: "\ue91b" is a valid path but we got an error: invalid net type.
    multiwallet_test.go:52: #47: failed on input "\ue91b"

Process finished with exit code 1
```
Analysis
- The test failed even though the path was valid. This shouldn't be.
- An error may be returned even when the path is valid. Therefore, it is incorrect to expect no errors simply because the path is valid.
- This the reason the last test run on this PR failed. It shouldn't have.

#### Other observations:
- It turns out that relative paths are supported but whether or not this is desired is yet to be seen.
- `canUseDir` creates the directory if it does not exist, should also ensure that the directory is deleted IF it was created. Should not delete the directory if it was already existing.

#### An interesting edge case
If the test code is corrected to use a valid `netType` (say `testnet3`), the test may still fail with the following error:
```
BAD: "" is NOT a valid path but we got no errors. Weird!
2020-01-22 12:20:57.101 [INF] DLWL: Loaded 0 wallets
    multiwallet_test.go:52: #90: failed on input ""

Process finished with exit code 1
```
This means that the `NewMultiWallet` function had no issues using the directory (no errors returned from the function) but `canUseDir` said the directory is not valid. Who's wrong?